website/pages/docs/plugins/sources/aws/event-based-sync.mdx

```diff
@@ -0,0 +1,91 @@
+
+AWS CloudTrail enables users to get an audit log of events occurring within their account. By subscribing to a stream of AWS CloudTrail events in a Kinesis Data stream CloudQuery can trigger selective syncs to update just the singular resource that had a configuration change.
+
+
+## Supported Services and Events
+Service | Event 
+-|-|
+ec2.amazonaws.com | AttachInternetGateway
+ec2.amazonaws.com | AuthorizeSecurityGroupEgress
+ec2.amazonaws.com | AuthorizeSecurityGroupIngress
+ec2.amazonaws.com | CreateInternetGateway
+ec2.amazonaws.com | CreateSecurityGroup
+ec2.amazonaws.com | CreateSubnet
+ec2.amazonaws.com | CreateTags
+ec2.amazonaws.com | CreateVpc
+ec2.amazonaws.com | DeleteTags
+ec2.amazonaws.com | DetachInternetGateway
+ec2.amazonaws.com | ModifySubnetAttribute
+ec2.amazonaws.com | RevokeSecurityGroupEgress
+ec2.amazonaws.com | RevokeSecurityGroupIngress
+ec2.amazonaws.com | RunInstances
+iam.amazonaws.com | CreateGroup
+iam.amazonaws.com | CreateRole
+iam.amazonaws.com | CreateUser
+rds.amazonaws.com | CreateDBCluster
+rds.amazonaws.com | CreateDBInstance
+
+
+## Configuration:
+
+1. Configure a AWS CloudTrail Trail to send management events to a Kinesis Data Stream via CloudWatch Logs. The most straight forward way to do this is to use the CloudQuery provided CloudFormation template.
+
+```
+aws cloudformation deploy --template-file ./streaming-deployment.yml --stack-name <STACK-NAME> --capabilities CAPABILITY_IAM --disable-rollback --region <DESIRED-REGION>
+```
+
+2. Copy the ARN of the Kinesis stream. If you used the CloudFormation template you can run the following command:
+```
+aws cloudformation describe-stacks --stack-name <STACK-NAME> --query "Stacks[].Outputs" --region <DESIRED-REGION>
+```
+
+3. Define a `config.yml` file like the one below
+
+``` yaml
+kind: source
+spec:
+  name: "aws-event-based"
+  registry: "local"
+  path: <PATH/TO/BINARY>
+  tables:
+    - aws_ec2_instances
+    - aws_ec2_internet_gateways
+    - aws_ec2_security_groups
+    - aws_ec2_subnets
+    - aws_ec2_vpcs
+    - aws_ecs_cluster_tasks
+    - aws_iam_groups
+    - aws_iam_roles
+    - aws_iam_users
+    - aws_rds_instances
+  destinations: ["postgresql"]
+  skip_tables:
+    - aws_iam_group_last_accessed_details
+    - aws_iam_role_last_accessed_details
+    - aws_iam_user_last_accessed_details
+  spec:
+    event_based_sync:
+      - account:
+          local_profile: "<ROLE-NAME>"
+        kinesis_stream_arn: <OUTPUT-FROM-CLOUDFORMATION-STACK>
+```
+
+4. Sync the data! 
+```
+cloudquery sync config.yml
+```
+
+This will start a long lived process that will only stop when there is an error or you stop the process
+
+
+### Limitations:
+- Kinesis Stream can only have a single shard. (This is a limitation that we expect to remove in the future)
+- Stale records will only be deleted if the plugin stops consuming the Kinesis Stream, which only can occur if there is an error
+- Incremental table's state is not updated by event based syncs
```

Comment:
```suggestion
- Incremental tables' state is not updated by event based syncs
```

Are any incremental tables supported by event-based syncs right now? Maybe we can remove this limitation?